pkg/pipelines/webhook/webhook.go

```diff
@@ -0,0 +1,223 @@
+package webhook
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/openshift/odo/pkg/pipelines/config"
+	"github.com/openshift/odo/pkg/pipelines/eventlisteners"
+	"github.com/openshift/odo/pkg/pipelines/git"
+	"github.com/openshift/odo/pkg/pipelines/ioutils"
+	"github.com/openshift/odo/pkg/pipelines/routes"
+	"github.com/openshift/odo/pkg/pipelines/secrets"
+)
+
+type webhookInfo struct {
+	clusterResource *resources
+	repository      *git.Repository
+	gitRepoURL      string
+	cicdNamepace    string
+	listenerURL     string
+	accessToken     string
+	names           []string
+	isCICD          bool
+}
+
+// Create creates a new webhook on the target Git Repository
+// names is a {envName, appName, seviceName} tuple.
+// It returns the ID of created webhook.
+func Create(accessToken, pipelinesFile string, names []string, isCICD bool) (string, error) {
```

Comment:
These could probably be pulled up into the `webhookInfo` type.

Reply:
This is the entry point from the CLI.   The CLI shoudn't need to know about webhookInfo.